x-pack/plugins/apm/server/routes/alerts/rule_types/transaction_duration/get_transaction_duration_chart_preview.ts

```diff
@@ -51,21 +55,31 @@ export async function getTransactionDurationChartPreview({
     transactionType,
     transactionName,
     interval,
-    start,
     end,
+    groupBy,
   } = alertParams;
   const searchAggregatedTransactions = await getSearchTransactionsEvents({
     config,
     apmEventClient,
     kuery: '',
   });
 
+  const intervalAsSeconds = getIntervalInSeconds(interval);
+  const intervalAsMs = intervalAsSeconds * 1000;
+  const start = end - intervalAsMs;
+
   const query = {
     bool: {
       filter: [
-        ...termQuery(SERVICE_NAME, serviceName),
-        ...termQuery(TRANSACTION_TYPE, transactionType),
-        ...termQuery(TRANSACTION_NAME, transactionName),
+        ...termQuery(SERVICE_NAME, serviceName, {
+          queryEmptyString: false,
+        }),
```

Comment:
Why do we want to query for empty service names?

Reply:
For the existing rules that might have an empty value. This is caused when user selected all we passed an empty string https://github.com/elastic/kibana/blob/8.6/x-pack/plugins/apm/public/components/alerting/utils/fields.tsx#L40-L44

I have fixed it in 8.8 https://github.com/elastic/kibana/blob/8.8/x-pack/plugins/apm/public/components/alerting/utils/fields.tsx#L44

Reply:
Ok, that makes sense. Thanks for the context! We might want to call this out explicitly - either with a code comment or better: a test that ensures that empty strings are allowed for BWC reasons.

Reply:
Added tests for empty fields 7ae99b1513b0619320a3be480f83801e9074e87b